### PR TITLE
[GH-16800]: Refactor raw SQL to query builder in group_store.go - part 1

### PR DESCRIPTION
**Pull request by djanda97**
#### Summary

This PR replaces raw SQL with the query builder for the following methods of group_store.go:
- Get
- Update
- Delete
- GetByRemoteID
- GetAllBySource
- Restore

#### Ticket Link

#16800

#### Release Note

```release-note

```

### CODE REVIEW

File: server/channels/store/sqlstore/group_store.go [hunk 1]
```diff
@@ -215,7 +215,12 @@ func (s *SqlGroupStore) buildInsertGroupUsersQuery(groupId string, userIds []str
 
 func (s *SqlGroupStore) Get(groupId string) (*model.Group, error) {
 	var group model.Group
-	if err := s.GetReplicaX().Get(&group, "SELECT * from UserGroups WHERE Id = ?", groupId); err != nil {
+	query, args, _ := s.getQueryBuilder().
```
Comment (isacikgoz) [inline server/channels/store/sqlstore/group_store.go:218]: What was the reason of ignoring errors here?

Comment (djanda97) [reply server/channels/store/sqlstore/group_store.go:218]: Oops, looks like I forgot to capture the error values 😅
I will capture them and add in early returns if an error is found.

File: server/channels/store/sqlstore/group_store.go [hunk 2]
```diff
@@ -261,7 +271,20 @@ func (s *SqlGroupStore) GetByIDs(groupIDs []string) ([]*model.Group, error) {
...
+			"RemoteId": remoteID,
+			"Source":   groupSource,
+		}).
+		ToSql()
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.GetReplicaX().Get(&group, query, args...); err != nil {
```
Comment (agnivade) [inline server/channels/store/sqlstore/group_store.go:287]: You can also simplify the whole thing as:

```
s.GetReplicaX().GetBuilder(&group, builder) 
```

and pass just the builder. This method will automatically convert to query and args and also return error if necessary. This gets rid of the repetetive boilerplate.

**Review by agnivade — COMMENTED** [review]
Just a small improvement. Looking good!

File: server/channels/store/sqlstore/group_store.go (hunk 2 above)
Comment (djanda97) [reply server/channels/store/sqlstore/group_store.go]: Thanks for the recommendation! I like the boilerplate reduction 😄

**Review by agnivade — APPROVED** [review]
Nice work!

File: server/channels/store/sqlstore/group_store.go [hunk 3]
```diff
@@ -362,7 +395,13 @@ func (s *SqlGroupStore) Delete(groupID string) (*model.Group, error) {
 
 func (s *SqlGroupStore) Restore(groupID string) (*model.Group, error) {
 	var group model.Group
-	if err := s.GetReplicaX().Get(&group, "SELECT * from UserGroups WHERE Id = ? AND DeleteAt != 0", groupID); err != nil {
+	builder := s.getQueryBuilder().
+		Select("*").
+		From("UserGroups").
+		Where(sq.Eq{"Id": groupID}).
+		Where(sq.NotEq{"DeleteAt": 0})
```
Comment (isacikgoz) [inline server/channels/store/sqlstore/group_store.go:402]: nit: I generally prefer as following but nothing blocking :)
```suggestion
		Where(sq.And{
			sq.Eq{"Id": groupID},
			sq.NotEq{"DeleteAt": 0},
		})

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks for the contribution @djanda97 🎉

**Comment by agnivade** [comment]
@toninis - The EE tests are not being triggered in this PR. Any idea why?

**Pull request merged**